### PR TITLE
feat: Added endpoint to update testimonials

### PR DESCRIPTION
**Pull request by theijhay**
​

## Description

Implemented the PATCH endpoint /api/v1/testimonials/{testimonial.id} to enable the updating of existing testimonials. This feature includes input validation and error handling to ensure that updates are processed correctly and that only valid testimonial data is accepted.

​

## Related Issue (Link to issue ticket)
This is the issue ticket #776 

​

## Motivation and Context

The update to this endpoint is necessary to allow users to modify testimonial information. This functionality enhances the flexibility of managing testimonials and ensures that the API can handle updates efficiently.

​

## How Has This Been Tested?

- Used Postman to test various update scenarios, including valid and invalid data.
- Verified that the endpoint correctly updates testimonials and handles errors as expected.
- Ran unit tests to confirm that the new functionality does not impact existing code.

Screenshot from 2024-08-07 16-45-36

​

## Screenshots (if appropriate - Postman, etc):

Screenshot from 2024-08-07 16-45-26

​

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: api/v1/routes/testimonial.py [hunk 1]
```diff
@@ -90,3 +91,21 @@ def create_testimonial(
         data={"id": testimonial.id}
     )
     return response
+
+
+@testimonial.patch("/{testimonial_id}", response_model=success_response)
+def update_testimonial(
+    testimonial_id: UUID,
```
Comment (Goketech) [inline api/v1/routes/testimonial.py:98]: change this to string

File: api/v1/routes/testimonial.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
 from typing import Annotated
 from api.utils.pagination import paginated_response
 from api.v1.models.testimonial import Testimonial
+from uuid import UUID
```
Comment (Goketech) [inline api/v1/routes/testimonial.py:18]: not needed

File: api/v1/routes/testimonial.py [hunk 3]
```diff
@@ -90,3 +91,21 @@ def create_testimonial(
...
+def update_testimonial(
+    testimonial_id: UUID,
+    testimonial_data: CreateTestimonial,
+    db: Session = Depends(get_db),
+    current_user: User = Depends(user_service.get_current_user)
+):
+    """
+    Endpoint to update testimonial by id
+    """
+    updated_testimonial = testimonial_service.update(db, current_user.id, str(testimonial_id), testimonial_data)
```
Comment (Goketech) [inline api/v1/routes/testimonial.py:106]: rather than the conversion here, just specify the type to string

**Review by Goketech — APPROVED** [review]

**Comment by theijhay** [comment]
@Goketech  Made changes

**Pull request closed without merging**